src/components/oif/vancouver/screen.vue

```diff
@@ -0,0 +1,275 @@
+</template>
+
+<script>    
+
+    import 'assets/css/ocp/2019/global/theme.scss'
+
+    import Event from './event.vue'
+    import moment from 'moment'
+
+    import { mapGetters } from 'vuex'    
+
+
+    export default {
+        computed: {
+            ...mapGetters({
+                schedule: 'schedule'
+            }),
+            roomStyle: function() {
+                return {
+                    'font-size': this.schedule.room.name.length >= 10 ? '3.5rem' : '4.5rem',
+                    'line-height': this.schedule.room.name.length >= 10 ? '3.5rem' : '4.5rem',                    
+                }
+            },
+            trackStyle: function() {
+                var color = '#ffffff';
+                if (this.schedule.state.track &&
+                    this.schedule.state.track.color) {
+                    color = this.schedule.state.track.color;
+                }
+                return { 'color': color, 'font-size': this.schedule.room.name.length >= 15 ? '2.5rem' : '2rem' }
+            },
+            summitScheduleUrl: function() {
+                return 'https://2022ocpglobal.fnvirtual.app/a/schedule'
+            }
+        },
+        methods: {
+            formatRoomName(name) {
+                if (name.match(/^\d/)) {
+                    return name.replace(/\s/g, '');
+                }
+                return name
+            },
+            scheduleIsOver() {
+                return !this.schedule.state.events.curr;
+            },
+            setRandomBackground() {
+                const el = document.body;
+                const random_letter = String.fromCharCode(Math.floor(Math.random() * 4) + 65).toUpperCase();
+                const previousBackground = Array.from(el.classList).find(className => className.startsWith('background-'));
+                if (previousBackground) {
+                    el.classList.remove(previousBackground);
+                }
+                el.classList.add(`background-${random_letter}`);
+            },
+            toggleBodyClass(addRemoveClass, className) {
+                const el = document.body;
+                if (addRemoveClass === 'addClass') {
+                    const previousBackground = Array.from(el.classList).find(className => className.startsWith('background-'));
+                    el.classList.remove(previousBackground);
+                    el.classList.add(className);
+                } else {
+                this.setRandomBackground();
+                el.classList.remove(className);
+                }
+            },
+            syncStart(item) {
+                this.schedule.setOffset(
+                    item.start_date - moment.utc().unix() - 65
+                )
+            },
+            syncEnd(item) {
+                this.schedule.setOffset(
+                    item.end_date - moment.utc().unix() - 5
+                )
+            },
+        },
+        mounted() {
+
+            if (!this.schedule.state.events.curr) {
```

Comment:
check computed property here

Reply:
also lets create a helper called
setBackground(showNoEvents){
if(showNoEvents) {
this.toggleBodyClass('addClass', 'background-no-events');
} else {
this.toggleBodyClass('removeClass', 'background-no-events');
this.setRandomBackground();
}
}